src/wizards/debugging/TaskCreator.ts

```diff
@@ -87,52 +113,67 @@ export class TaskCreator {
+    taskDefinition['label'] = taskLabel;
+    taskDefinition['command'] = unixCommand;
+    taskDefinition['windows'] = { command: windowsCommand };
+    taskDefinition['problemMatcher'][0]['background']['beginsPattern'] = taskPattern.beginsPattern;
+    taskDefinition['problemMatcher'][0]['background']['endsPattern'] = taskPattern.endsPattern;
+
+    if (!FsUtils.isSameDirectory(this.workspaceFolder.uri.fsPath, this.projectFolder)) {
+      taskDefinition.options = { cwd: path.relative(this.workspaceFolder.uri.fsPath, this.projectFolder) };
+    }
+
+    return taskDefinition;
+  }
+
+  /**
+   * Returns the task definition for starting Quarkus development mode
+   *
+   * @returns the task definition for starting Quarkus development mode
+   */
+  private async getQuarkusDevTask(): Promise<TaskDefinition> {
     const taskLabel: string = this.quarkusBuildSupport.getQuarkusDevTaskName(this.workspaceFolder, this.projectFolder);
     const unixCommand: string = (await this.quarkusBuildSupport.getQuarkusDevCommand(this.projectFolder, { windows: false })).command;
     const windowsCommand: string = (await this.quarkusBuildSupport.getQuarkusDevCommand(this.projectFolder, { windows: true })).command;
-
     const taskPatterns: TaskPattern = this.quarkusBuildSupport.getTaskPatterns();
-    const task: TaskDefinition =
-    {
-      label: taskLabel,
-      type: "shell",
-      command: unixCommand,
-      windows: {
-        command: windowsCommand
-      },
-      isBackground: true,
-      problemMatcher: [
-        {
-          pattern: [
-            {
-              regexp: "\\b\\B",
-              file: 1,
-              location: 2,
-              message: 3
-            }
-          ],
-          background: {
-            activeOnStart: true,
-            beginsPattern: taskPatterns.beginsPattern,
-            endsPattern: taskPatterns.endsPattern,
-          }
-        }
-      ]
-    };
-
-    if (!FsUtils.isSameDirectory(this.workspaceFolder.uri.fsPath, this.projectFolder)) {
-      task.options = {cwd: path.relative(this.workspaceFolder.uri.fsPath, this.projectFolder)};
-    }
+    return this.getTaskFromCommandLine(taskLabel, unixCommand, windowsCommand, taskPatterns);
+  }
 
-    return task;
+  /**
+   * Returns the task definition for building a native image of the Quarkus app
+   *
+   * @returns the task definition for building a native image of the Quarkus app
+   */
+  private async getQuarkusBinaryTask(): Promise<TaskDefinition> {
+    const taskLabel: string = this.quarkusBuildSupport.getQuarkusBinaryTaskName(this.workspaceFolder, this.projectFolder);
+    const unixCommand: string = (await this.quarkusBuildSupport.getQuarkusBinaryCommand(this.projectFolder, { windows: false })).command;
+    const windowsCommand: string = (await this.quarkusBuildSupport.getQuarkusBinaryCommand(this.projectFolder, { windows: true })).command;
+    const taskPatterns: TaskPattern = this.quarkusBuildSupport.getTaskPatterns();
```

Comment:
This specifies to use the same TaskPatterns as for quarkus:dev, but the build tool output is different between quarkus:dev and the binary build

Reply:
Removed task patterns for build executable